theatre/studio/src/StudioStore/StudioStore.ts

```diff
@@ -141,46 +142,62 @@ export default class StudioStore {
               )
 
-              forEachDeep(
-                value,
-                (v, pathToProp) => {
-                  if (typeof v === 'undefined' || v === null) {
-                    return
-                  }
-
-                  const propAddress = {...root.address, pathToProp}
-
-                  const trackId = get(
-                    sequenceTracksTree,
-                    pathToProp,
-                  ) as $FixMe as SequenceTrackId | undefined
-
-                  if (typeof trackId === 'string') {
-                    const propConfig = get(
-                      root.template.config.props,
-                      pathToProp,
-                    ) as PropTypeConfig | undefined
-                    if (propConfig?.sanitize) v = propConfig.sanitize(v)
-
-                    const seq = root.sheet.getSequence()
-                    seq.position = seq.closestGridPosition(seq.position)
-                    stateEditors.coreByProject.historic.sheetsById.sequence.setKeyframeAtPosition(
-                      {
-                        ...propAddress,
-                        trackId,
-                        position: seq.position,
-                        value: v as $FixMe,
-                        snappingFunction: seq.closestGridPosition,
-                      },
-                    )
-                  } else {
-                    stateEditors.coreByProject.historic.sheetsById.staticOverrides.byObject.setValueOfPrimitiveProp(
-                      {...propAddress, value: v},
-                    )
-                  }
-                },
-                getPointerParts(pointer as Pointer<$IntentionalAny>).path,
-              )
+              const propConfig = get(
+                root.template.config.props,
+                path,
+              ) as PropTypeConfig
+
+              const setStaticOrKeyframeProp = <T>(
+                value: T,
+                path: PathToProp,
+              ) => {
+                if (typeof value === 'undefined' || value === null) {
+                  return
+                }
+
+                const propAddress = {...root.address, pathToProp: path}
+
+                const trackId = get(sequenceTracksTree, path) as $FixMe as
+                  | SequenceTrackId
+                  | undefined
+                if (typeof trackId === 'string') {
+                  const propConfig = get(root.template.config.props, path) as
+                    | PropTypeConfig
+                    | undefined
+                  if (propConfig?.sanitize) value = propConfig.sanitize(value)
+
+                  const seq = root.sheet.getSequence()
+                  seq.position = seq.closestGridPosition(seq.position)
+                  stateEditors.coreByProject.historic.sheetsById.sequence.setKeyframeAtPosition(
+                    {
+                      ...propAddress,
+                      trackId,
+                      position: seq.position,
+                      value: value as $FixMe,
```

Comment:
Can't trust that user won't mutate `value`. Perhaps we should clone it?

Reply:
... because with the rgba type, `value` is no longer just `string | number | boolean`

Reply:
Cloning already happens both for the default value, and when interpolating, and I think this kind of stupid proofing should only happen on library boundaries anyway, when accepting, or when providing a value. Tbh the rgba object *is* cloned here when we call sanitize on it, so it does in fact happen here.

Reply:
Oh, the API boundary *is* the set function, silly me. I'm going to clone right at the beginning.